Sources/NIOPosix/RawSocketBootstrap.swift

```diff
@@ -191,5 +192,306 @@ public final class NIORawSocketBootstrap {
     }
 }
 
+// MARK: Async connect/bin methods with NIOAsyncChannel
+
+extension NIORawSocketBootstrap {
+    /// Bind the `Channel` to `host`.
+    /// All packets or errors matching the `ipProtocol` specified are passed to the resulting `Channel`.
+    ///
+    /// - Parameters:
+    ///   - host: The host to bind on.
+    ///   - ipProtocol: The IP protocol used in the IP protocol/nextHeader field.
+    ///   - backpressureStrategy: The back pressure strategy used by the channel.
+    ///   - isOutboundHalfClosureEnabled: Indicates if half closure is enabled on the channel. If half closure is enabled
+    ///   then finishing the `NIOAsyncChannelWriter` will lead to half closure.
+    ///   - inboundType: The channel's inbound type.
+    ///   - outboundType: The channel's outbound type.
+    /// - Returns: A `NIOAsyncChannel` for the bound connection.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @_spi(AsyncChannel)
+    public func bind<Inbound: Sendable, Outbound: Sendable>(
```

Comment:
Should we be providing the initialiser for this (and `connect`) here?

We have a split world where you can specify it in the initialiser on the bootstrap and in some cases (protocol negotiation and the "return whatever you want from the initializer") in `bind`/`connect`. 

It might be easier if all the `async` `bind`/`connect` methods have initialisers.

Reply:
Yeah realised this today as well but we gotta revisit it holistically. I will defer this to the overall API review.

Reply:
Did we not add them with inits in the other NIO bootstraps though?

Reply:
Nope, the `NIOAsyncChannel` based ones are all without an explicit initialiser